app/controllers/admin/payments_controller.rb

```diff
@@ -133,26 +133,37 @@ def create_invoice(csv, payments, payment_type, start_date, end_date = nil)
         p = Participant.where(member: payment.member, activity_id: activity_id).first
         # now create every transaction row with the following date: Blank, ledger account,
         # transaction description, VAT number, amount, cost_location ()
-        csv <<  if p.activity.group.nil? || Group.first.ledgernr.blank?
-                  ["", "1302", "#{ p.activity.name } - #{ p.member_id }", p.activity.VAT,
-                   p.currency, ""]
-                else
-                  ["", p.activity.group.ledgernr, "#{ p.activity.name } - #{ p.member_id }",
-                   p.activity.VAT, p.currency, p.activity.group.cost_location]
-                end
+        csv << if Activity.where(id: activity_id).description == "Lidmaatschap"
+                 # Dislike this way of doing it but need better ways to find membership activities
+                 if p.activity.group.nil? || Group.first.ledgernr.blank?
+                   ["", "1302", "#{ p.activity.name } - #{ p.member_id }", p.activity.VAT,
+                    p.currency + payment.transaction_fee, ""]
+                 else
+                   ["", p.activity.group.ledgernr, "#{ p.activity.name } - #{ p.member_id }",
+                    p.activity.VAT, p.currency + payment.transaction_fee, p.activity.group.cost_location]
+                 end
+               elsif p.activity.group.nil? || Group.first.ledgernr.blank?
+                 ["", "1302", "#{ p.activity.name } - #{ p.member_id }", p.activity.VAT,
+                  p.currency, ""]
+               else
+                 ["", p.activity.group.ledgernr, "#{ p.activity.name } - #{ p.member_id }",
+                  p.activity.VAT, p.currency, p.activity.group.cost_location]
+               end
       end
     end
 
     payments.where(transaction_type: :checkout).group(:member_id).sum(:amount).each do |payment|
```

Comment:
Wouldn't it be better not to group by member_id here? if someone top ups multiple time per week, they will get multiple lines of csv, but that's fine.
It keeps this code a lot simpler in my opinion.

Reply:
Totally agree, was over complicating stuff here